### ISSUE TITLE
Update status does not return number of documents for failures

### ISSUE THREAD

**Issue by bidoubiwa** [meilisearch/MeiliSearch#1517]
**Describe the bug**
Previously all updates linked to documents: `documentsAddition/Partial/Deletions`  always returned the number of documents impacted in all steps of the update. In v0.21 it is only returned upon successful addition, thus status `processed`. In all other scenarios: `enqueued`, `processing` and `failed` it is not returned

v0.20
```js
  update: {
    status: 'enqueued',
    updateId: 0,
    type: { name: 'DocumentsAddition', number: 19547 }, // number provided
    enqueuedAt: '2021-07-19T09:39:35.150281Z'
  }
```

```js
{
  {
        "status": "failed",
        "updateId": 3,
        "type": {
            "name": "DocumentsAddition",
            "number": 6 // number provided
        },
// ...
}
```

v0.21:

// enqueued
```js
{
  update: {
    status: 'enqueued',
    updateId: 0,
    type: { name: 'DocumentsAddition' },             // number not provided
    enqueuedAt: '2021-07-19T09:38:11.238813Z'
  }
}
```

// Processing 
```js
{
  update: {
    status: 'processing',
    updateId: 0,
    type: { name: 'DocumentsAddition' },             // number not provided
    enqueuedAt: '2021-07-19T09:38:42.100731Z',
    startedProcessingAt: '2021-07-19T09:38:42.101528Z'
  }
}
```

// Failed
```js
   {
        "status": "failed",
        "updateId": 4,
        "type": {
            "name": "DocumentsAddition"
            // number not provided
        },
}
```

**To Reproduce**
1. Index big dataset
2. Get updates, should be processing
3. Index big dataset again
4. Get updates
5. should be enqueued
6. Remove data.ms
7. Index dataset with a document without `primary_key`
8. Get updates
8. Should have failed

**Expected behavior**
keep `number` in the returned update or explain breaking behavior

**MeiliSearch version:** 
v0.21

**Comment by bb** [comment]
I prefer keeping the number (and/or other easy-to-implement metadata) because it helps identifying issues with my indexing code.

**Comment by qdequele** [comment]
Indeed I think it is a problem, thank you for pointing it out Charlotte. @gmourier I let you give your opinion and approve we want to keep the behavior on the v0.21. If it's approved, can @MarinPostma or @irevoire do it?

**Comment by gmourier** [comment]
Yes! I think the 0.20 behavior is more logical and clearer, it would be great to keep it. @marin is it something complicated to fix for 0.21?

**Comment by bb** [comment]
Thanks for the inside view.
For me it's not important enough to justify a dirty hack.

**Comment by Kerollmops** [comment]
@MarinPostma milli always [returns the number of indexed documents](https://github.com/meilisearch/milli/blob/16698f714b7d9d25b1ee75deff0e0d704cf9dae1/milli/src/update/index_documents/mod.rs#L329) so you could at least display it when an update has been successfully indexed. Am I right?

**Comment by bb** [comment]
That works already 👍

**Comment by gmourier** [comment]
I think we'll have to fix it during Q3 when we implement csv / ndjson if it's not possible to do it now other than with a dirty hack. This is a regression that can be significant. We shouldn't lose this information if people are using it.

**Comment by bb** [comment]
I was just sending a rather large DocumentsAddition request, about 750000 documents at once and wondered if it would be a good addition to add the Payload size (to know if I'm close to the Limit). What do you think?

**Comment by curquiza** [comment]
Checked with the team: unfortunately, this is not easy to fix in a clean way and we don't want to delay the v0.21.0 for this. We are aware this is not convenient for the users and we are sorry for the inconvenience, we will re-integrate it in the next versions.
FYI, the number of documents is still available if the update succeeds!

**Comment by curquiza** [comment]
Fixed in v0.25.0 with the new task API -> the number of documents is present even for failed task
